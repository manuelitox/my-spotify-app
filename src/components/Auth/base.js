import React, { Fragment } from 'react'

import theme from 'config/theme'
import Title from 'components/ArtistPage/Title/'
import PopUp from 'components/generic/PopUp/container'
import StyledPopUp from './styles'

const AuthPopUp = () => (
  <PopUp isOpen={ true } togglePopUp={ () => console.log('') } WithCloseButton={ false }>
    <Fragment>
      <Title color={ theme.blueDark } insidePopUp={ true }>Spotify Authorization</Title>
      <StyledPopUp>
        <a href={ `https://accounts.spotify.com/authorize?client_id=${ process.env.REACT_APP_CLIENT_ID }&redirect_uri=${ encodeURIComponent(process.env.REACT_APP_URL) }&scope=user-read-private%20user-read-email&response_type=token&state=123` }>Get Authorization</a>
        <p><small>You need to login into Spotify</small></p>
      </StyledPopUp>
    </Fragment>
  </PopUp>            
)

export default AuthPopUp