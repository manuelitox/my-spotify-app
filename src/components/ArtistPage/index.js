import React from 'react'
 
import Header from './Header/'
import Banner from './Banner/'
import Albums from './Albums/'
import Tracks from './Tracks/'
import RelatedArtists from './RelatedArtists/'
import StyledArtistPage from './styles'
import StyledContainerAlbumsAndTopTracks from './stylesContainerAlbumsAndTopTracks'

const ArtistPage = () => (
  <StyledArtistPage>
    <Header />
    <Banner />
    <StyledContainerAlbumsAndTopTracks>
      <Albums />
      <Tracks />
    </StyledContainerAlbumsAndTopTracks>
    <RelatedArtists />
  </StyledArtistPage>
)

export default ArtistPage