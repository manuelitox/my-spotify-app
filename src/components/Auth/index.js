import React, { Component } from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'

import PopUp from './base'
import GetUrl from 'lib/auth/get-url'
import GetToken from 'lib/auth/get-token'
import { authUpdate } from 'actions/auth'

export class Auth extends Component {
  componentDidMount() {
    const { authUpdate } = this.props
    const url = GetUrl(window.location.href)
    const token = GetToken(url)
    authUpdate(token)
  }

  render() { 
    const { token } = this.props
    return !token ? <PopUp /> : null
  }
}

Auth.propTypes = {
  token: PropTypes.string,
  authUpdate: PropTypes.func.isRequired
}

const mapStateToProps = (state) => ({
  token: state.AuthReducer.token
})

export default connect(
  mapStateToProps,
  { authUpdate }
)(Auth)