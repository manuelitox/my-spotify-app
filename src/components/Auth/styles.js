import styled from 'styled-components'

const StyledPopUp = styled.div`
  a {
    background-color: ${ props => props.theme.SpotifyColor };
    border-radius: 4px;
    color: white;
    display: inline-block;
    padding: 10px 20px;
  }
`

export default StyledPopUp