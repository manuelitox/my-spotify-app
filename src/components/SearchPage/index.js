import React, { Fragment } from 'react'

import SearchFormContainer from 'components/generic/SearchForm/container'
import StyledSearchPage from './styles'
import StyledWaveSearchIcon from './stylesWaveIcon'
import { ReactComponent as HeadsetIcon } from 'svgs/headset.svg'

const SearchPage = () => (
  <Fragment>
    <StyledSearchPage>
      <HeadsetIcon />
      <SearchFormContainer />
    </StyledSearchPage>
    <StyledWaveSearchIcon />
  </Fragment>
)

export default SearchPage