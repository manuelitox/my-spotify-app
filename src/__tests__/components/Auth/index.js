import React from 'react'
import { shallow } from 'enzyme'

import { Auth } from 'components/Auth/'

describe('Component | Auth', () => {
  let component 
  beforeEach( () => {
    component = shallow(<Auth authUpdate={ jest.fn() } />)
  })
  it('should render a component', () => {
    expect(component).toMatchSnapshot()
  })
  describe('Token exists', () => {
    beforeEach( () => {
      component = shallow(<Auth authUpdate={ jest.fn() } token="123123" />)
    })    
    it('should not render Auth component', () => {
      expect(component).toMatchSnapshot()
    })
  })
})