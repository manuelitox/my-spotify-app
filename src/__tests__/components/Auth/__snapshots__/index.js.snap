// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component | Auth Token exists should not render Auth component 1`] = `""`;

exports[`Component | Auth should render a component 1`] = `<AuthPopUp />`;
